android/src/main/java/com/revenuecat/purchases/common/PurchasesPeriod.java

```diff
@@ -84,24 +87,22 @@ static PurchasesPeriod parse(@NonNull CharSequence text) {
                     int days = parseNumber(text, dayMatch, negate);
                     days = PurchasesMath.addExact(days, PurchasesMath.multiplyExact(weeks, 7));
                     return create(years, months, days);
-                } catch (NumberFormatException ex) {
-                    throw new RuntimeException("Text cannot be parsed to a Period: " + text, ex);
+                } catch (NumberFormatException | ArithmeticException ex) {
+                    Log.e("PURCHASES", "Text cannot be parsed to a Period: " + text);
+                    return null;
                 }
             }
         }
-        throw new RuntimeException("Text cannot be parsed to a Period: " + text);
+        Log.e("PURCHASES", "Text cannot be parsed to a Period: " + text);
+        return null;
     }
 
     private static int parseNumber(CharSequence text, String str, int negate) {
         if (str == null) {
             return 0;
         }
         int val = Integer.parseInt(str);
-        try {
-            return PurchasesMath.multiplyExact(val, negate);
-        } catch (ArithmeticException ex) {
-            throw new RuntimeException("Text cannot be parsed to a Period: " + text, ex);
-        }
+        return PurchasesMath.multiplyExact(val, negate);
```

Comment:
this, would throw `ArithmeticException` which will be capture by the catch in `parse`